include/bluetooth/mesh/scheduler_cli.h

```diff
@@ -0,0 +1,173 @@
+		       const struct bt_mesh_scheduler_status *status);
+
+	/** @brief Scheduler Action Status message callback.
+	 *
+	 *  @param[in] cli    Scheduler client model.
+	 *  @param[in] ctx    Message context parameters.
+	 *  @param[in] status Received Scheduler Action Status.
+	 */
+	void (*action_status)(struct bt_mesh_scheduler_cli *cli,
+			struct bt_mesh_msg_ctx *ctx,
+			const struct bt_mesh_scheduler_action_status *status);
+
+	/* Composition data entry pointer. */
+	struct bt_mesh_model *mod;
+	/* Model publication parameters. */
+	struct bt_mesh_model_pub pub;
+	/* Publication message */
+	struct net_buf_simple pub_msg;
+	/* Publication message buffer. */
+	uint8_t buf[BT_MESH_MODEL_BUF_LEN(BT_MESH_SCHEDULER_OP_ACTION_STATUS,
+			BT_MESH_SCHEDULER_MSG_LEN_ACTION_STATUS)];
+	/* Ack context */
+	struct bt_mesh_model_ack_ctx ack;
+	/* Transaction ID */
+	uint8_t tid;
+};
+
+/** @brief Get the current Schedule Register status.
+ *
+ *  This call is blocking if the @c rsp buffer is non-NULL. The response will
+ *  always be passed to the @ref bt_mesh_scheduler_cli::status callback.
+ *
+ *  @param[in]  cli Scheduler client model.
+ *  @param[in]  ctx Message context to send with, or NULL to use the configured
+ *                  publication parameters.
+ *  @param[out] rsp Response buffer, or NULL to keep from blocking.
+ *
+ *  @retval 0 Successfully got the Schedule Register status.
+ *  @retval -EALREADY A blocking request is already in progress.
+ *  @retval -EADDRNOTAVAIL A message context was not provided and publishing is
+ *  not configured.
+ *  @retval -EAGAIN The device has not been provisioned.
+ *  @retval -ETIMEDOUT The request timed out without a response.
+ */
+int bt_mesh_scheduler_cli_get(struct bt_mesh_scheduler_cli *cli,
+			      struct bt_mesh_msg_ctx *ctx,
+			      struct bt_mesh_scheduler_status *rsp);
+
+/** @brief Get the appropriate Scheduler Action status.
+ *
+ *  This call is blocking if the @c rsp buffer is non-NULL. The response will
+ *  always be passed to the @ref bt_mesh_scheduler_cli::action_status callback.
+ *
+ *  @param[in]  cli   Scheduler client model.
+ *  @param[in]  ctx   Message context to send with, or NULL to use
+ *                    the configured publication parameters.
+ *  @param[in]  index Index of the Schedule Register entry to get.
+ *  @param[out] rsp   Response buffer, or NULL to keep from blocking.
+ *
+ *  @retval 0 Successfully got the Scheduler Action status.
+ *  @retval -EINVAL Invalid index.
+ *  @retval -EALREADY A blocking request is already in progress.
+ *  @retval -EADDRNOTAVAIL A message context was not provided and publishing is
+ *  not configured.
+ *  @retval -EAGAIN The device has not been provisioned.
+ *  @retval -ETIMEDOUT The request timed out without a response.
+ */
+int bt_mesh_scheduler_cli_action_get(struct bt_mesh_scheduler_cli *cli,
+				struct bt_mesh_msg_ctx *ctx,
+				uint8_t index,
+				struct bt_mesh_scheduler_action_status *rsp);
```

Comment:
It is parsed data from the incoming message. If everything works well they are the same but it will skip malfunction behavior if a server replies to the wrong index.